File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int count = 0; // 칠 횟수 카운트
        int location = 0; // 롤러 위치
        
        // section을 탐색하면서
        for (int i : section) {
            // 롤러의 위치보다 보수해야 할 위치가 뒤에있으면
            if (i > location) {
                count++; // 칠 횟수를 증가시키고
                location = i + m - 1; // 롤러의 위치 업데이트
            }
        }
        return count;
    }
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    var count = 0; // 칠 횟수 카운트
    var location = 0; // 롤러의 마지막 위치
    
    // 칠이 필요한 구역 탐색
    for (let i = 0; i < section.length; i++) {
        // 만약 롤러의 위치보다 칠해야 하는 구역이 더 뒤에있으면 작업한다
        if (section[i] > location) {
            count++; // 칠 횟수 추가
            location = section[i] + m - 1; // 롤러의 마지막 위치 갱신
        }
    }
    
    return count;
}